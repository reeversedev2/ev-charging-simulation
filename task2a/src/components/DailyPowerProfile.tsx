import { Zap } from 'lucide-react';
import {
	Area,
	AreaChart,
	CartesianGrid,
	Legend,
	Line,
	ResponsiveContainer,
	Tooltip,
	XAxis,
	YAxis,
} from 'recharts';

const generateMockDailyData = () => {
	return Array.from({ length: 24 }, (_, hour) => {
		const baseLoad = 20 + Math.sin((hour * Math.PI) / 12) * 15;
		const randomVariation = Math.random() * 10;
		return {
			hour: `${hour.toString().padStart(2, '0')}:00`,
			power: Math.max(0, Math.min(baseLoad + randomVariation, 100)),
			efficiency: Math.random() * 30 + 60,
		};
	});
};

const DailyPowerProfile = () => {
	const mockDailyData = generateMockDailyData();

	return (
		<div
			className="bg-white rounded-lg shadow-sm border border-gray-200 w-full"
			data-testid="daily-power-profile"
		>
			<div className="p-4 border-b border-gray-200">
				<h2 className="flex items-center gap-2 text-lg font-semibold">
					<Zap className="w-5 h-5" /> Daily Power Profile
				</h2>
			</div>
			<div className="p-4">
				<div className="h-80">
					<ResponsiveContainer width="100%" height="100%">
						<AreaChart data={mockDailyData}>
							<defs>
								<linearGradient id="powerGradient" x1="0" y1="0" x2="0" y2="1">
									<stop offset="5%" stopColor="#3b82f6" stopOpacity={0.8} />
									<stop offset="95%" stopColor="#3b82f6" stopOpacity={0.1} />
								</linearGradient>
							</defs>
							<CartesianGrid strokeDasharray="3 3" />
							<XAxis dataKey="hour" />
							<YAxis yAxisId="left" />
							<YAxis yAxisId="right" orientation="right" />
							<Tooltip />
							<Legend />
							<Area
								yAxisId="left"
								type="monotone"
								dataKey="power"
								stroke="#3b82f6"
								fill="url(#powerGradient)"
								name="Power (kW)"
							/>
							<Line
								yAxisId="right"
								type="monotone"
								dataKey="efficiency"
								stroke="#10b981"
								name="Efficiency (%)"
							/>
						</AreaChart>
					</ResponsiveContainer>
				</div>
			</div>
		</div>
	);
};

export default DailyPowerProfile;
