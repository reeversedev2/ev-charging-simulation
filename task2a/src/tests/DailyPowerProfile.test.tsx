import { render, screen } from '@testing-library/react';
import DailyPowerProfile from '../components/DailyPowerProfile';
import { describe, expect, it } from 'vitest';

describe('DailyPowerProfile', () => {
	it('renders the component title correctly', () => {
		render(<DailyPowerProfile />);
		expect(screen.getByText('Daily Power Profile')).toBeInTheDocument();
	});

	it('matches snapshot', () => {
		const { container } = render(<DailyPowerProfile />);
		expect(container).toMatchSnapshot();
	});

	it('renders with correct styling classes', () => {
		render(<DailyPowerProfile />);
		expect(screen.getByTestId('daily-power-profile')).toHaveClass(
			'bg-white',
			'rounded-lg'
		);
	});
});
