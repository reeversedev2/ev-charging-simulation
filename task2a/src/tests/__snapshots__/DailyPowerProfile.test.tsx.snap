// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DailyPowerProfile > matches snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-lg shadow-sm border border-gray-200 w-full"
    data-testid="daily-power-profile"
  >
    <div
      class="p-4 border-b border-gray-200"
    >
      <h2
        class="flex items-center gap-2 text-lg font-semibold"
      >
        <svg
          class="lucide lucide-zap w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
          />
        </svg>
         Daily Power Profile
      </h2>
    </div>
    <div
      class="p-4"
    >
      <div
        class="h-80"
      >
        <div
          class="recharts-responsive-container"
          style="width: 100%; height: 100%; min-width: 0;"
        />
      </div>
    </div>
  </div>
</div>
`;
